packages/osd-i18n/src/core/locales.js

```diff
@@ -36,3 +36,4 @@ addLocaleData({ locale: "ko", pluralRuleFunction: function (n,ord){if(ord)return
 addLocaleData({ locale: "zh-CN", parentLocale: "zh" });
+addLocaleData({ locale: "demo", pluralRuleFunction: function (n,ord){if(ord)return"other";return"other"},"fields":{"year":{"displayName":"年","relative":{"0":"今年","1":"明年","-1":"去年"},"relativeTime":{"future":{"other":"{0}年后"},"past":{"other":"{0}年前"}}},"year-short":{"displayName":"年","relative":{"0":"今年","1":"明年","-1":"去年"},"relativeTime":{"future":{"other":"{0}年后"},"past":{"other":"{0}年前"}}},"month":{"displayName":"月","relative":{"0":"本月","1":"下个月","-1":"上个月"},"relativeTime":{"future":{"other":"{0}个月后"},"past":{"other":"{0}个月前"}}},"month-short":{"displayName":"月","relative":{"0":"本月","1":"下个月","-1":"上个月"},"relativeTime":{"future":{"other":"{0}个月后"},"past":{"other":"{0}个月前"}}},"day":{"displayName":"日","relative":{"0":"今天","1":"明天","2":"后天","-2":"前天","-1":"昨天"},"relativeTime":{"future":{"other":"{0}天后"},"past":{"other":"{0}天前"}}},"day-short":{"displayName":"日","relative":{"0":"今天","1":"明天","2":"后天","-2":"前天","-1":"昨天"},"relativeTime":{"future":{"other":"{0}天后"},"past":{"other":"{0}天前"}}},"hour":{"displayName":"小时","relative":{"0":"这一时间 \u002F 此时"},"relativeTime":{"future":{"other":"{0}小时后"},"past":{"other":"{0}小时前"}}},"hour-short":{"displayName":"小时","relative":{"0":"这一时间 \u002F 此时"},"relativeTime":{"future":{"other":"{0}小时后"},"past":{"other":"{0}小时前"}}},"minute":{"displayName":"分钟","relative":{"0":"此刻"},"relativeTime":{"future":{"other":"{0}分钟后"},"past":{"other":"{0}分钟前"}}},"minute-short":{"displayName":"分","relative":{"0":"此刻"},"relativeTime":{"future":{"other":"{0}分钟后"},"past":{"other":"{0}分钟前"}}},"second":{"displayName":"秒","relative":{"0":"现在"},"relativeTime":{"future":{"other":"{0}秒钟后"},"past":{"other":"{0}秒钟前"}}},"second-short":{"displayName":"秒","relative":{"0":"现在"},"relativeTime":{"future":{"other":"{0}秒后"},"past":{"other":"{0}秒前"}}}} });
```

Comment:
Can we add this behind a flag? Not sure if its a good idea to register a demo locale for everyone

Reply:
sorry. what do you mean by behind a flag?

Reply:
Currently based on my understanding, the demo locale is always available to every user. That shouldn't be the case right?

Reply:
I went back and read the issue, but I'm still not sure I understand why it's necessary to actually register a demo locale. Is there other code that explicitly references `demo` as the locale name? If it's just for example or illustration purposes, it could just be added directly to the readme.

Reply:
@ashwin-pc @joshuarrrr  for the first concern, I am discussing with the team on it


"The reason why I add a demo locale is because currently I name the translation example file as demo.json [here](https://github.com/opensearch-project/i18n-plugin/tree/main/translations) . I think we have two options 1) keep the demo locale. pros: the name is very clear on our responsibility. we won’t have a lot of maintenance issue beside the demo file. Cons: we add one more locale  2) we change i18n-plugin/translation/demo.json to zh-CN.json. Pros: we don’t include another locale Cons: name is a bit ambiguous on what is our maintenance responsibility. "

any thoughts on this?

Reply:
That's helpful, thanks. So both `demo.json` and the registration of the the `demo` locale are intended to just help developers see what they need to do to add a new translation.

I think @ashwin-pc is arguing that we shouldn't register `demo` by default, but only when in some sort of development or example mode. Rather than making a new feature flag or configuration setting for this, I wonder if it would be possible to rely on the same `--run-examples` flag used to activate the other developer examples. Can plugins like this access the env CLI option? If so, I think that registering only in that mode would be a nice solution.

I don't think we should make the demo file `zh-CN.json`, because I agree that it's confusing, unless it's a real `zh-CN` localization file we intend to support as maintainers.

Also, very minor, but I think I'd prefer `example` instead of `demo`, just because it seems a bit more consistent with other similar features.

Reply:
sounds reasonable to me. I can research on this to see if there is some development mode that we could pass an argument to register a locale.